com.unity.netcode.gameobjects/Tests/Runtime/DistributedAuthority/SessionVersionConnectionRequest.cs

```diff
@@ -0,0 +1,106 @@
+using System.Collections;
+using NUnit.Framework;
+using Unity.Netcode.TestHelpers.Runtime;
+using UnityEngine.TestTools;
+
+namespace Unity.Netcode.RuntimeTests
+{
+    internal class SessionVersionConnectionRequest : NetcodeIntegrationTest
+    {
+        protected override int NumberOfClients => 0;
+
+        public SessionVersionConnectionRequest() : base(NetworkTopologyTypes.DistributedAuthority, HostOrServer.DAHost) { }
+
+        private bool m_UseValidSessionVersion;
+        private bool m_ClientWasDisconnected;
+        private NetworkManager m_ClientNetworkManager;
+
+        /// <summary>
+        /// Callback used to mock the scenario where a client has an invalid session version
+        /// </summary>
+        /// <returns><see cref="SessionConfig"/></returns>
+        private SessionConfig GetInavlidSessionConfig()
```

Comment:
It might also be good to have a test on the new client connecting with `SessionVersion + 1` and checking that they get downgraded?

Reply:
I thought about that but realized I would need to replicate that code in the NGO SDK to get a DAHost to handle this...which would sort of defeat that particular test.
The downgrade check should be a test that is validated in the service specific integration test that should verify this same existing functionality in this test plus the downgrade.